parsley/shared/src/main/scala/parsley/internal/deepembedding/frontend/IterativeEmbedding.scala

```diff
@@ -10,15 +10,21 @@ import scala.collection.{mutable, Factory}
 import parsley.internal.deepembedding.ContOps, ContOps.{suspend, ContAdapter}
 import parsley.internal.deepembedding.backend, backend.StrictParsley
 
-private [parsley] final class Many[A, C](init: LazyParsley[mutable.Builder[A, C]], p: =>LazyParsley[A], private [parsley] var debugName: String)
-    extends Binary[mutable.Builder[A, C], A, C](init, p) {
+/**
+  * This trait represents combinators that are iterative, that is, they execute 
+  * parsers multiple times until they cannot match any more
+  */
+sealed trait Iterative
```

Comment:
still needs `private [parsley]`, since we don't want this exposed in public API.